protocol/synthetix/contracts/storage/Pool.sol

```diff
@@ -420,27 +418,24 @@ library Pool {
         return Market.load(0);
     }
 
-    function getRequiredMinDelegationTime(
-        Data storage self
-    ) internal view returns (uint32 requiredMinDelegateTime) {
+    function getRequiredDelegationDelayAndWindow(
+        Data storage self,
+        bool isUndelegation
+    ) internal view returns (uint32 requiredDelayTime, uint32 requiredWindowTime) {
         for (uint256 i = 0; i < self.marketConfigurations.length; i++) {
-            uint32 marketMinDelegateTime = Market
-                .load(self.marketConfigurations[i].marketId)
-                .minDelegateTime;
-
-            if (marketMinDelegateTime > requiredMinDelegateTime) {
-                requiredMinDelegateTime = marketMinDelegateTime;
+            Market.Data storage market = Market.load(self.marketConfigurations[i].marketId);
+            uint32 marketDelayTime = isUndelegation
+                ? market.undelegateCollateralDelay
+                : market.delegateCollateralDelay;
+
+            if (marketDelayTime > requiredDelayTime) {
+                requiredDelayTime = marketDelayTime;
+                // Also get the window from the more restrictive market
+                requiredWindowTime = isUndelegation
```

Comment:
Am not sure does more restrictive window 🤔 ... if it's most strict based on delay, if that's the intention i dont see an issue with it... but maybe fix the comment. I understand the complexity if a market has a 0 window, it means it's less strict that other window configurations, but we have the edge case of 2 year window, and 0 which cause confusion, since 1 year is less strict than 2 years...  For that reason, i'm ok keeping it as is i feel.

Reply:
done